doc/services/storage/zms/zms.rst

```diff
@@ -0,0 +1,432 @@
+pairs until it is full.
+
+The key-value pair is divided into two parts:
+
+- The key part is written in an ATE (Allocation Table Entry) called "ID-ATE" which is stored
+  starting from the bottom of the sector
+- The value part is defined as "DATA" and is stored raw starting from the top of the sector
+
+Additionally, for each sector we store at the last positions Header-ATEs which are ATEs that
+are needed for the sector to describe its status (closed, open) and the current version of ZMS.
+
+When the current sector is full we verify first that the following sector is empty, we garbage
+collect the N+2 sector (where N is the current sector number) by moving the valid ATEs to the
+N+1 empty sector, we erase the garbage collected sector and then we close the current sector by
+writing a garbage_collect_done ATE and the close ATE (one of the header entries).
+Afterwards we move forward to the next sector and start writing entries again.
+
+This behavior is repeated until it reaches the end of the partition. Then it starts again from
+the first sector after garbage collecting it and erasing its content.
+
+Composition of a sector
+=======================
+A sector is organized in this form (example with 3 sectors):
+
+.. list-table::
+   :widths: 25 25 25
+   :header-rows: 1
+
+   * - Sector 0 (closed)
+     - Sector 1 (open)
+     - Sector 2 (empty)
+   * - Data_a0
+     - Data_b0
+     - Data_c0
+   * - Data_a1
+     - Data_b1
+     - Data_c1
+   * - Data_a2
+     - Data_b2
+     - Data_c2
+   * - GC_done
+     -    .
+     -    .
+   * -    .
+     -    .
+     -    .
+   * -    .
+     -    .
+     -    .
+   * -    .
+     - ATE_b2
+     - ATE_c2
+   * - ATE_a2
+     - ATE_b1
+     - ATE_c1
+   * - ATE_a1
+     - ATE_b0
+     - ATE_c0
+   * - ATE_a0
+     - GC_done
+     - GC_done
+   * - Close (cyc=1)
+     - Close (cyc=1)
+     - Close (cyc=1)
+   * - Empty (cyc=1)
+     - Empty (cyc=2)
+     - Empty (cyc=2)
+
+Definition of each element in the sector
+========================================
+
+``Empty ATE:`` is written when erasing a sector (last position of the sector).
+
+``Close ATE:`` is written when closing a sector (second to last position of the sector).
+
+``GC_done ATE:`` is written to indicate that the next sector has been already garbage
+collected. This ATE could be in any position of the sector.
+
+``ID-ATE:`` are entries that contain a 32 bits Key and describe where the data is stored, its
+size and its crc32
+
+``Data:`` is the actual value associated to the ID-ATE
+
+How does ZMS work?
+******************
+
+Mounting the Storage system
+===========================
+
+Mounting the storage starts by getting the flash parameters, checking that the file system
+properties are correct (sector_size, sector_count ...) then calling the zms_init function to
+make the storage ready.
+
+To mount the NVS filesystem some elements in the zms_fs structure must be initialized.
```

Comment:
```suggestion
To mount the filesystem some elements in the zms_fs structure must be initialized.
```
`NVS` is used multiple times in this document. The use in this location is inaccurate and can be skipped

Reply:
Fixed